lib/util/fonts/getTextByFontProperties.js

```diff
@@ -684,8 +685,42 @@ function getTextByFontProperties(htmlAsset, memoizedGetCssRulesByProperty) {
                 idArray = [0];
             }
 
-            // Inputs might have visual text, but don't have childNodes
-            if (node.tagName === 'INPUT' && visualValueInputTypes.indexOf(node.type || 'text') !== -1) {
+            if (node.tagName === 'NOSCRIPT') {
+                htmlAsset.outgoingRelations.some(function (relation) {
+                    if (relation.type === 'HtmlNoscript' && relation.node === node) {
+                        noscriptStack.push(true);
+                        var noscriptDocument = relation.to.parseTree;
+                        var isWithinBody = false;
+                        for (var i = 0 ; i < noscriptDocument.childNodes.length ; i += 1) {
```

Comment:
This next block seems like a lot of gynmastics to work against our document markers. Not a nice user experience to duplicate any time you need to traverse a noscript tag. Could we somehow encapsulate this in the parseTree getter?

Reply:
Yeah, you're right. I think this boilerplate stems from the fact that I copy/pasted from HtmlConditionalComment, which needs to be able to contain unbalanced HTML, `<html>` and other weirdness. I'll see if it can be simplified :+1:

Reply:
Done (+ updated https://github.com/assetgraph/assetgraph/pull/787), take another look now :)